nimble/include/nimble/nimble_npl.h

```diff
@@ -172,6 +175,20 @@ ble_npl_time_t ble_npl_time_ms_to_ticks32(uint32_t ms);
 BLE_NPL_STORAGE_DECL
 uint32_t ble_npl_time_ticks_to_ms32(ble_npl_time_t ticks);
 
+typedef void (*ble_npl_task_func_t)(void *);
+struct ble_npl_task;
+
+BLE_NPL_STORAGE_DECL
+int ble_npl_task_init(struct ble_npl_task *t, const char *name, ble_npl_task_func_t func,
+		 void *arg, uint8_t prio, ble_npl_time_t sanity_itvl,
+		 ble_npl_stack_t *stack_bottom, uint16_t stack_size);
```

Comment:
we do not have tasks in NPL on purpose - required tasks should be created by target os/application (we'll have this described in doc)

the reason is that there only a handful of tasks required depending on what features should be enabled. the API above is copied straight from Mynewt and is not really portable. for example, FreeRTOS API does allocate stack internally instead of having it passed as a parameter. so unless there is good API proposal for this I'd stick with our current approach.

Reply:
Nimble is creating tasks internally, specifically `nimble/transport/socket/src/ble_hci_socket.c`, so should I sought to use npl for this.  FreeRTOS10 can be configured to take passed in pre-allocated stack.  I think npl should provide a complete osal abstraction, which would include a task api.

Reply:
tasks are only created once and "just work" so for now this is done by app with provided task entry function:
https://github.com/apache/mynewt-nimble/blob/master/nimble/controller/src/ble_ll.c#L1543
https://github.com/andrzej-kaczmarek/apache-mynewt-nimble/blob/f4d909e2d6c0624b8936963e5dd204a219ae78cd/porting/examples/freertos_nrf5_sdk/main.c#L118

we can add task creation abstraction later (other APIs for tasks are not needed) as I'd prefer to have this defined in some better way than just c&p from Mynewt to cover differences between OS-es.

Reply:
In the case of hci sock, there are order-of-operation issues with your proposed technique.  main can't safely create the hci task before or after calling `ble_hci_sock_init`.  

If called before, event queue isn't setup.

If called after, config fails (as it tries to open socket and setup hci before event handler task is started).

Both failures result in failure setting up socket:
```
bind() failed 19
nimble-linux: ../../../nimble/transport/socket/src/ble_hci_socket.c:852: ble_hci_sock_init: Assertion `Failure configuring socket HCI' failed.
Aborted (core dumped)
```